src/sql/src/session/vars.rs

```diff
@@ -903,7 +903,16 @@ const STORAGE_DATAFLOW_MAX_INFLIGHT_BYTES: ServerVar<Option<usize>> = ServerVar
     internal: true,
 };
 
-/// The fraction of the cluster replica size to be used as the maximum number of
+/// Whether or not to delay sources producing values in some scenarios
+/// (namely, upsert) till after rehydration is finished.
+const STORAGE_DATAFLOW_DELAY_SOURCES_PAST_REHYDRATION: ServerVar<bool> = ServerVar {
+    name: UncasedStr::new("storage_dataflow_delay_sources_past_rehydration"),
+    value: &true,
```

Comment:
default should be `false` to keep existing behaviour same?

Reply:
The behavior should be safe, i added this flag so we can turn it off if there is a problem!

Reply:
Echoing the request to make this off by default! It can probably be a `feature_flag!` below too which will save you some typing.

Reply:
ill probably leave it here to avoid churn